tests/TestFramework.re

```diff
@@ -0,0 +1,21 @@
+module MockSnapshotIO = {
+  let readSnapshotNames = _ => [];
+  let readFile: string => string = s => s;
+  let removeFile: string => unit = _ => ();
+  let writeFile: (string, string) => unit = (_, _) => ();
+  let existsFile: string => bool = _ => false;
+  let mkdirp = _ => ();
+};
+/* Snapshots will be virtual-modulified at some point, until then we need to inject
+ * something custom with RelyInternal to make JSOO happy because the default snapshot
+ * implementation uses the Unix module  */
```

Comment:
I may be missing something about rely - do we need to have snapshotting work with JSOO?  Testing would not happen in the browser

Reply:
Snapshots aren't dependent on the browser, basically it just checks some string output versus a file and provides a hook to create/update the file if it doesn't exist/the test fails. You may not need them at all for testing irmin, but for things like code gen/pretty printers/anything that produces really long string output it's a nice feature to have that I want to make sure eventually works